speckle/base/client.py

```diff
@@ -118,6 +118,26 @@ def login(self, email, password):
             'Authorization': self.me['token'],
         })
 
+    def login_with_token(self, token):
+        """Login user to the speckle server using an API token
+
+        If your server uses an alternate login mechansim (e.g. ActiveDirectory), you can use this function to
+        store your API token in the client.
+
+        Arguments:
+            token {str} -- your Speckle API token (format 'JWT xxxxxxx', found on the Speckle web interface)"""
+        
+        if type(token) is not str:
+            raise ValueError("Token must be a string")
+
+        if (token[0:3] != 'JWT'):
+            raise ValueError("Toekn must begin with 'JWT'")
+        
+        self.s.headers.update({
+            'content-type': 'application/json',
+            'Authorization': token,
+        })
+    
```

Comment:
this looks like a useful addition. thanks a lot David! it's definitely better to have something like this in place rather than a weird hack that might get lost in the forums.

I have a couple of lil notes:

1. there's a typo (Toekn -> Token 😅) 
2. I would probably add the token to the `me` dict at `self.me['token']` since it is used elsewhere and is passed to the resources which leads me to believe that stuff might break if the token isn't there. @tsvilans might have more insight

also a small suggestion - not a huge deal but I would probably check the 'JWT' start as you are, but just add it to the token if it is missing as a small convenience to the user rather than raising a value error. eg:

```py
if not token.startswith('JWT'):
    token = f'JWT {token}'
```

Reply:
Thanks @izzylys. I've fixed the typo and added in the token the `self.me` dict. I've made a search through the code, and it seems that me['token'] is the only data that is actually accessed (even though other info gets stored).

I'm not comfortable with automatically adding `JWT ` to the token. That could be helpful at a user interface level, but for an API we should prioritize predictability, and this feels like a bit of magic. I added the check to point devs to the right direction, but since there are an infinite number of ways that the token can be screwed up, I am not comfortable with a 'magic' solution that assumes a particular type of error.